WordPress/gutenberg
# Try/fake file templates

## Description

- Overload the REST controllers of templates and template parts to return both `publish` posts **and theme files** (only if there is no corresponding custom post already).
- To be included in the response, theme files are converted into "fake" `WP_Post` objects, with a temporary ID (`file-template-${ slug }`). These fake posts are **not saved in the database**.
- When the site editor request template/parts, it simply adds them to the entity store as usual. Fake posts are inserted with that `file-template-` ID, but they work nonetheless.
- In the site editor, I've added a `TemplateProvider` wrapper for the normal `EntityProvider`. It observes the entity ID and, if it's `file-template-`, it **saves the fake post** into an actual auto draft, and updates the entity ID to the saved one (which is now a real post).
- Just to clarify: auto drafts are **only** created when a template/part becomes the main editor's entity. This happens automatically for the front page template on site editor load, or whenever we change template from the sidebar.
This **will create more auto drafts than strictly needed**, but to be honest that's not that much different than what happens when creating a new post.
- Now that we have a real ID, we can modify the template and publish it. Next requests will use the `publish` template, and won't include its corresponding theme file.
- I've also updated the Template Part block to use the correct fake template part, although I haven't tested template parts thouroughly yet.
- In the PHP side, I've updated the template resolution to also consider the theme files if no relevant `publish` post is found.

## How has this been tested?

- Checkout this PR first. This will disable the auto-sync of templates and parts.
- Delete all templates and parts to start clean. (It's not necessary, but it makes things clearer).
- Open the site on the front end, and make sure it renders the correct template and its template parts.
- Open the site editor, and observe the network tab in the dev tools.
  - Make sure the template and its template parts show up.
  - Make sure the template is saved as auto-draft, and the editor URL is updated using its real ID.
  - Edit the template and save it. Make sure it's now published.
  - Open another template, and make sure it's saved as auto-draft.
  - Open a template part, and make sure it's saved as auto-draft.
- Exit the editor and check the Templates and Template Parts lists in wp-admin.
  - Make sure they contain: 1 published template; 1 auto-draft template; 1 auto-draft template part.

## Screenshots 

## Types of changes





## Checklist:
